hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/HddsUtils.java

```diff
@@ -824,4 +826,23 @@ public static Map<String, String> processForLogging(OzoneConfiguration conf) {
     }
     return sortedOzoneProps;
   }
+
+  /**
+   * Execute some code and ensure thread name is not changed
+   * (workaround for HADOOP-18433).
+   */
+  public static <T, E extends IOException> T keepingThreadName(
```

Comment:
Minor nit
```suggestion
  public static <T, E extends IOException> T preserveThreadName(
```

Reply:
Thanks @kerneltime for the review.  Patch is updated.